main/src/ca/uwaterloo/flix/language/phase/jvm/GenExpression.scala

```diff
@@ -517,13 +517,105 @@ object GenExpression {
+
+      //Invoke the getRecordWithField method on the record. (To get the proper record object)
+      visitor.visitMethodInsn(INVOKEINTERFACE, interfaceType.name.toInternalName, "getRecordWithField",
+        AsmOps.getMethodDescriptor(List(JvmType.String), interfaceType), true)
+
+      //Cast to proper record extend class
+      visitor.visitTypeInsn(CHECKCAST, classType.name.toInternalName)
+
+      //Invoke the getField method on the record. (To get the proper value)
+      visitor.visitMethodInsn(INVOKEVIRTUAL, classType.name.toInternalName, "getField",
+        AsmOps.getMethodDescriptor(Nil, JvmOps.getErasedJvmType(tpe)), false)
+
+    case Expression.RecordExtend(label, value, rest, tpe, loc) =>
+      // Adding source line number for debugging
+      addSourceLine(visitor, loc)
+      // We get the JvmType of the class for the record extend
+      val classType = JvmOps.getRecordExtendClassType(tpe)
+
+      // We get the JvmType of the record interface
+      val interfaceType = JvmOps.getRecordInterfaceType()
+
+      // Instantiating a new object of tuple
+      visitor.visitTypeInsn(NEW, classType.name.toInternalName)
+      // Duplicating the class
+      visitor.visitInsn(DUP)
+
+      //Push the required argument to call the RecordExtend constructor.
+
+      //Push the label of field (which is going to be the extension).
+      visitor.visitLdcInsn(label)
+
+      //Push the value of the field onto the stack, since it is an expression we first need to compile it.
+      compileExpression(value, visitor, currentClass, lenv0, entryPoint)
+
+      //Push the value of the rest of the record onto the stack, since it's an expression we need to compile it first.
+      compileExpression(rest, visitor, currentClass, lenv0, entryPoint)
+
+      // Descriptor of constructor
+      val constructorDescriptor = AsmOps.getMethodDescriptor(List(JvmType.String, JvmOps.getErasedJvmType(value.tpe),
+        interfaceType), JvmType.Void)
+      // Invoking the constructor
+      visitor.visitMethodInsn(INVOKESPECIAL, classType.name.toInternalName, "<init>", constructorDescriptor, false)
+
+    case Expression.RecordRestrict(label, rest, tpe, loc) =>
+      // Adding source line number for debugging
+      addSourceLine(visitor, loc)
+      // We get the JvmType of the record interface
+      val interfaceType = JvmOps.getRecordInterfaceType()
 
-    case Expression.RecordSelect(exp, label, tpe, loc) => ???
+      //Push the value of the rest of the record onto the stack, since it's an expression we need to compile it first.
+      compileExpression(rest, visitor, currentClass, lenv0, entryPoint)
+      //Push the label of field (which is going to be the removed/restricted).
+      visitor.visitLdcInsn(label)
 
-    case Expression.RecordExtend(label, value, rest, tpe, loc) => ??? // TODO
+      // Invoking the removeField method
+      visitor.visitMethodInsn(INVOKEINTERFACE, interfaceType.name.toInternalName, "removeField",
```

Comment:
Can we rename it to `restrictField` just to use the same terminology throughout?